composer/models/base_hparams.py

```diff
@@ -87,3 +88,40 @@ def initialize_object(self) -> ComposerModel:
             ComposerModel: The constructed :class:`.ComposerModel`
         """
         pass
+
+
+@dataclass
+class TransformerHparams(ModelHparams, ABC):
```

Comment:
Should this be in its own file, and perhaps be called `HuggingfaceTransformerHparams`? From a quick glance, it seems to be pretty specific to the architecture of a huggingface transformer.

Reply:
After talking to @moinnadeem I'm going to break all the nlp stuff out into its own pr/design. It's more of a redesign than a refactor to align it with the cv parts right now.